Data/Config.hs

```diff
@@ -159,6 +161,49 @@ loadConfig path
              Left e    -> throwM e
              Right reg -> return $ Config reg
 
+--------------------------------------------------------------------------------
+-- | Loads a config without flattening any property object out.
+loadNotInlinedConfig :: (MonadIO m, MonadThrow m) => FilePath -> m Config
+loadNotInlinedConfig path
+    = do txt <- liftIO $ T.readFile path
+         pse <- parse path txt
+         let action
+                 = do ps         <- pse
+                      (tys, pts) <- typecheck $ rename ps
+                      return $ register tys pts
+         case action of
+             Left e    -> throwM e
+             Right reg -> return . Config . removeInlining . simplifyReg $ reg
```

Comment:
I think this could be simplified to:

```suggestion
    = Config . removeInlining . simplifyReg . unConf <$> loadConfig path
```

Also, I don't need and wouldn't add such a top-level function, but just use `removeInlining` and `simplifyReg`.

Reply:
Fair enough. We could also expose a single function that's simply the composition of `removeInlining` and `simplifyReg`.

Reply:
Sounds good to me!